site/webapp/src/main/java/uk/nhs/hee/web/eform/behaviors/StoreBlogCommentBehavior.java

```diff
@@ -34,9 +34,19 @@
     @Override
     public void onValidationSuccess(HstRequest request, HstResponse response, ComponentConfiguration config, FormBean bean, Form form, FormMap map) {
         try {
-            HstComponentConfiguration componentConfiguration = (HstComponentConfiguration) config.getClass().getMethod("getComponentConfiguration").invoke(config);
-            HstComponentConfiguration blogConfiguration = componentConfiguration.getParent().getParent().getChildByName("blog").getChildByName("blogpost");
-            String documentPath = blogConfiguration.getParameter("document");
+            // In the case of archived blog posts wherein the hst:relativecontentpath is set
+            // in the (non-workspace) sitemap
+            String documentPath = request.getRequestContext().getResolvedSiteMapItem().getRelativeContentPath();
```

Comment:
Would this be neater to extract into a getBlogDocumentPath method to hide the complexity?

Reply:
Done